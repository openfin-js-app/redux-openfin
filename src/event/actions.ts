import {EventConstant, appEvents, windowEvents} from './constants';

export type ActionDict = {
    [name:string]:EventConstant,
}

const appEventDictByName:ActionDict = {};
const appEventDictByType:ActionDict = {};
const windowEventDictByName:ActionDict = {};
const windowEventDictByType:ActionDict = {};

appEvents.forEach(oneEvent => {
    appEventDictByName[oneEvent.name] = oneEvent;
    appEventDictByType[oneEvent.type] = oneEvent;
});

windowEvents.forEach(oneEvent => {
    windowEventDictByName[oneEvent.name] = oneEvent;
    windowEventDictByType[oneEvent.type] = oneEvent;
});

export default {
    appEventDictByName,
    appEventDictByType,
    windowEventDictByName,
    windowEventDictByType,
};