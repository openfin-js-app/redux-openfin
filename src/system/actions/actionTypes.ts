import { makeReqType, makeResType }  from '../../utils/makeType';

export const GET_MACHINE_ID = makeReqType('GET_MACHINE_ID');
export const GET_MACHINE_ID_RES = makeResType('GET_MACHINE_ID_RES');

export const GET_DEVICE_USER_ID = makeReqType('GET_DEVICE_USER_ID');
export const GET_DEVICE_USER_ID_RES = makeResType('GET_DEVICE_USER_ID_RES');

export const GET_MONITOR_INFO = makeReqType('GET_MONITOR_INFO');
export const GET_MONITOR_INFO_RES = makeResType('GET_MONITOR_INFO_RES');

export const GET_VERSION = makeReqType('GET_VERSION');
export const GET_VERSION_RES = makeResType('GET_VERSION_RES');

export const GET_HOST_SPECS = makeReqType('GET_HOST_SPECS');
export const GET_HOST_SPECS_RES = makeResType('GET_HOST_SPECS_RES');

export const GET_ENVIRONMENT_VARIABLE = makeReqType('GET_ENVIRONMENT_VARIABLE');
export const GET_ENVIRONMENT_VARIABLE_RES = makeResType('GET_ENVIRONMENT_VARIABLE_RES');

export const CLEAR_CACHE = makeReqType('CLEAR_CACHE');
export const CLEAR_CACHE_RES = makeResType('CLEAR_CACHE_RES');