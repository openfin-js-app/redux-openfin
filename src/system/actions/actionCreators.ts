import createFSA, {ActionCreator} from '../../utils/createFSA';
import * as types from "../types";
import * as actionTypes from "./actionTypes";

export const getMachineId:ActionCreator<types.GetMachineIdPayload> = createFSA<types.GetMachineIdPayload>(
    actionTypes.GET_MACHINE_ID,
    (options:types.GetMachineIdPayload) => <any> options
);

export const getDeviceUserId:ActionCreator<types.GetDeviceUserIdPayload> = createFSA<types.GetDeviceUserIdPayload>(
    actionTypes.GET_DEVICE_USER_ID,
    (options:types.GetDeviceUserIdPayload) => <any> options
);

export const getMonitorInfo:ActionCreator<types.GetMonitorInfoPayload> = createFSA<types.GetMonitorInfoPayload>(
    actionTypes.GET_MONITOR_INFO,
    (options:types.GetMonitorInfoPayload) => <any> options
);

export const getVersion:ActionCreator<types.GetVersionPayload> = createFSA<types.GetVersionPayload>(
    actionTypes.GET_VERSION,
    (options:types.GetVersionPayload) => <any> options
);

export const getHostSpecs:ActionCreator<types.GetHostSpecsPayload> = createFSA<types.GetHostSpecsPayload>(
    actionTypes.GET_HOST_SPECS,
    (options:types.GetHostSpecsPayload) => <any> options
);


export const getEnvironmentVariable:ActionCreator<types.GetEnvironmentVariablePayload> = createFSA<types.GetEnvironmentVariablePayload>(
    actionTypes.GET_ENVIRONMENT_VARIABLE,
    (options:types.GetEnvironmentVariablePayload) => <any> options
);

export const clearCache:ActionCreator<types.ClearCachePayload> = createFSA<types.ClearCachePayload>(
    actionTypes.CLEAR_CACHE,
    (options:types.ClearCachePayload) => <any> options
);