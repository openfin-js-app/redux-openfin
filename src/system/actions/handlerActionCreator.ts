import {Action} from 'redux-actions';
import createFSA, {ActionCreator} from "../../utils/createFSA";
import * as types from "../types";
import * as actionTypes from "./actionTypes";

export const getMachineIdRes:ActionCreator<types.GetMachineIdResPayload> = createFSA<types.GetMachineIdResPayload>(
    actionTypes.GET_MACHINE_ID_RES,
    (options:types.GetMachineIdResPayload) => <any> options
);

export const getDeviceUserIdRes:ActionCreator<types.GetDeviceUserIdResPayload> = createFSA<types.GetDeviceUserIdResPayload>(
    actionTypes.GET_DEVICE_USER_ID_RES,
    (options:types.GetDeviceUserIdResPayload) => <any> options
);

export const getMonitorInfoRes:ActionCreator<types.GetMonitorInfoResPayload> = createFSA<types.GetMonitorInfoResPayload>(
    actionTypes.GET_MONITOR_INFO_RES,
    (options:types.GetMonitorInfoResPayload) => <any> options
);

export const getVersionRes:ActionCreator<types.GetVersionResPayload> = createFSA<types.GetVersionResPayload>(
    actionTypes.GET_VERSION_RES,
    (options:types.GetVersionResPayload) => <any> options
);

export const getHostSpecsRes:ActionCreator<types.GetHostSpecsResPayload> = createFSA<types.GetHostSpecsResPayload>(
    actionTypes.GET_HOST_SPECS_RES,
    (options:types.GetHostSpecsResPayload) => <any> options
);


export const getEnvironmentVariableRes:ActionCreator<types.GetEnvironementVariableResPaylod> = createFSA<types.GetEnvironementVariableResPaylod>(
    actionTypes.GET_ENVIRONMENT_VARIABLE_RES,
    (options:types.GetEnvironementVariableResPaylod) => <any> options
);

export const clearCacheRes:ActionCreator<types.ClearCacheResPayload> = createFSA<types.ClearCacheResPayload>(
    actionTypes.CLEAR_CACHE_RES,
    (options:types.ClearCacheResPayload) => <any> options
);