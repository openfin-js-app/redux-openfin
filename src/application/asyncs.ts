import {Action} from 'redux-actions';
import * as types from "./types";
import * as handlerActions from './actions/handlerActionCreator';

import createAsyncFun from '../utils/createAsyncFun'

import {
    NEW_APPLICAITON_ERROR_MSG,
    RESTART_ERROR_MSG,
    CLOSE_ERROR_MSG,
} from './types';

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#Application
export async function newApplication(action:Action<types.NewApplicationPayload>):Promise<Action<types.NewApplicationResPayload>>{
    const  options  = action.payload;
    return createAsyncFun<types.NewApplicationPayload,types.NewApplicationResPayload>(
        action,
        NEW_APPLICAITON_ERROR_MSG,
        handlerActions.newApplicationRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            let app = new fin.desktop.Application(options,
                (app:any)=>{
                    const responseAction = resActionCreator({app});
                    succCb(responseAction);
                },errCb);
        }
    );
}

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#.getCurrent

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getWindow

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#.wrap

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#addEventListener

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#close
export async function close(action:Action<types.ClosePayload>):Promise<Action<types.CloseResPayload>>{
    const { force } = action.payload;
    return createAsyncFun<types.ClosePayload,types.CloseResPayload>(
        action,
        CLOSE_ERROR_MSG,
        handlerActions.closeRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            let application = fin.desktop.Application.getCurrent();
            application.close(force,
                ()=>{
                    const responseAction = resActionCreator({});
                    succCb(responseAction);
                },errCb);
        }
    );
}

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getChildWindows

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getInfo

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getShortcuts

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getTrayIconInfo

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getZoomLevel

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#isRunning

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#removeEventListener

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#restart
export async function restart(action:Action<types.RestartPayload>):Promise<Action<types.RestartResPayload>>{
    return createAsyncFun<types.RestartPayload,types.NewApplicationResPayload>(
        action,
        RESTART_ERROR_MSG,
        handlerActions.restartRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            let application = fin.desktop.Application.getCurrent();
            application.restart(
                ()=>{
                    const responseAction = resActionCreator({});
                    succCb(responseAction);
                },errCb);
        }
    );
}

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#run

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#scheduleRestart

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#setShortcuts

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#setTrayIcon

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#setZoomLevel

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#terminate
