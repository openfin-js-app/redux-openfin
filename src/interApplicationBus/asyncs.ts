import {Action} from 'redux-actions';
import * as types from "./types";
import * as handlerActions from './actions/handlerActionCreator';

import wrapAsyncFun from '../utils/wrapAsyncFun'

export async function publish(action:Action<types.PublishPayload>):Promise<Action<types.PublishResPayload>>{
    const  { topic, message }  = action.payload;
    return wrapAsyncFun<types.PublishPayload,types.PublishResPayload>(
        action,
        handlerActions.publishRes,
        async (fin)=>{
            await fin.InterApplicationBus.publish(topic, message);
            return handlerActions.publishRes({});
        }
    );
}

export async function subscribe(action:Action<types.SubscribePayload>):Promise<Action<types.SubscribeResPayload>>{
    const  { source, topic, listener }  = action.payload;
    return wrapAsyncFun<types.SubscribePayload,types.SubscribeResPayload>(
        action,
        handlerActions.subscribeRes,
        async (fin)=>{
            await fin.InterApplicationBus.subscribe(source, topic, listener);
            return handlerActions.subscribeRes({});
        }
    );
}
