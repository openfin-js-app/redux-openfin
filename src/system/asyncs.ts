import {Action} from 'redux-actions';
import * as types from "./types";
import * as handlerActions from './actions/handlerActionCreator';

import createAsyncFun from '../utils/createAsyncFun'

import {
    GET_MACHINE_ID_ERROR_MSG,
    GET_DEVICE_USER_ID_ERROR_MSG,
    GET_MONITOR_INFO_ERROR_MSG,
    GET_VERSION_ERROR_MSG,
    GET_HOST_SPECS_ERROR_MSG,
    GET_ENVIRONMENT_VARIABLE_ERROR_MSG,
    CLEAR_CACHE_ERROR_MSG,
} from './types';

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.System.html#.getDeviceId
export async function getDeviceId(action:Action<types.GetMachineIdPayload>):Promise<Action<types.GetMachineIdResPayload>>{
    return createAsyncFun<types.GetMachineIdPayload,types.GetMachineIdResPayload>(
        action,
        GET_MACHINE_ID_ERROR_MSG,
        handlerActions.getMachineIdRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            fin.desktop.System.getMachineId(
                (id:string)=>{
                    const responseAction = resActionCreator({id});
                    succCb(responseAction);
                },
                errCb
            );
        }
    );
}

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.System.html#.getDeviceUserId
export async function getDeviceUserId(action:Action<types.GetDeviceUserIdPayload>):Promise<Action<types.GetDeviceUserIdResPayload>>{
    return createAsyncFun<types.GetDeviceUserIdPayload,types.GetDeviceUserIdResPayload>(
        action,
        GET_DEVICE_USER_ID_ERROR_MSG,
        handlerActions.getDeviceUserIdRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            fin.desktop.System.getDeviceUserId(
                (id:string)=>{
                    const responseAction = resActionCreator({id});
                    succCb(responseAction);
                },
                errCb
            );
        }
    );
}

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.System.html#.getMonitorInfo
export async function getMonitorInfo(action:Action<types.GetMonitorInfoPayload>):Promise<Action<types.GetMonitorInfoResPayload>>{
    return createAsyncFun<types.GetMonitorInfoPayload,types.GetMonitorInfoResPayload>(
        action,
        GET_MONITOR_INFO_ERROR_MSG,
        handlerActions.getMonitorInfoRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            fin.desktop.System.getMonitorInfo(
                (monitorInfo:any)=>{
                    const responseAction = resActionCreator(monitorInfo);
                    succCb(responseAction);
                },
                errCb
            );
        }
    );
}

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.System.html#.getVersion
export async function getVersion(action:Action<types.GetVersionPayload>):Promise<Action<types.GetVersionResPayload>>{
    return createAsyncFun<types.GetVersionPayload,types.GetVersionResPayload>(
        action,
        GET_VERSION_ERROR_MSG,
        handlerActions.getVersionRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            fin.desktop.System.getVersion(
                (version:string)=>{
                    const responseAction = resActionCreator({version});
                    succCb(responseAction);
                },
                errCb
            );
        }
    );
}

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.System.html#.getHostSpecs
export async function getHostSpecs(action:Action<types.GetHostSpecsPayload>):Promise<Action<types.GetHostSpecsResPayload>>{
    return createAsyncFun<types.GetHostSpecsPayload,types.GetHostSpecsResPayload>(
        action,
        GET_HOST_SPECS_ERROR_MSG,
        handlerActions.getHostSpecsRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            fin.desktop.System.getHostSpecs(
                (info:types.GetHostSpecsResPayload)=>{
                    const responseAction = resActionCreator(info);
                    succCb(responseAction);
                },
                errCb
            );
        }
    );
}


//http://cdn.openfin.co/jsdocs/beta/fin.desktop.System.html#.getEnvironmentVariable
export async function getEnvironmentVariable(action:Action<types.GetEnvironmentVariablePayload>)
    :Promise<Action<types.GetEnvironementVariableResPaylod>>
{
    const { env } = action.payload;
    return createAsyncFun<types.GetEnvironmentVariablePayload,types.GetEnvironementVariableResPaylod>(
        action,
        GET_ENVIRONMENT_VARIABLE_ERROR_MSG,
        handlerActions.getEnvironmentVariableRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            fin.desktop.System.getEnvironmentVariable(
                env,
                (value:string)=>{
                    const responseAction = resActionCreator({env,value});
                    succCb(responseAction);
                },
                errCb
            );
        }
    );
}

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.System.html#.clearCache
export async function clearCache(action:Action<types.ClearCachePayload>):Promise<Action<types.ClearCacheResPayload>>{
    const  options  = action.payload;
    return createAsyncFun<types.ClearCachePayload,types.ClearCacheResPayload>(
        action,
        CLEAR_CACHE_ERROR_MSG,
        handlerActions.clearCacheRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            fin.desktop.System.clearCache(
                options,
                ()=>{
                    const responseAction = resActionCreator({});
                    succCb(responseAction);
                },
                errCb
            );
        }
    );

}