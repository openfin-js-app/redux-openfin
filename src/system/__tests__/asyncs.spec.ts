jest.mock('../../utils/createAsyncFun');
const createAsyncFun = require('../../utils/createAsyncFun');
import * as actions from '../actions';
import * as asyncs  from '../asyncs';

describe('System asyncs',()=>{

    afterEach(()=>{
        jest.resetAllMocks();
    });

    it('getMachineId',async ()=>{
        const getMachineId = jest.fn((succCb,errCb)=>{
            succCb();
        });
        const fin = {
            desktop:{
                System:{
                    getMachineId
                }
            }
        };
        const succCb = jest.fn();
        const errCb = jest.fn();
        createAsyncFun.default=jest.fn(
            (action,ERROR_MSG,resActionCreator,finCb)=>{
                finCb(fin,action,resActionCreator,succCb,errCb);
            }
        );
        await asyncs.getMachineId(
            actions.getMachineId({})
        );
        expect(getMachineId).toMatchSnapshot();
        expect(succCb).toMatchSnapshot();
        expect(createAsyncFun).toMatchSnapshot();
    });

    it('getDeviceUserId',async ()=>{
        const getDeviceUserId = jest.fn((succCb,errCb)=>{
            succCb();
        });
        const fin = {
            desktop:{
                System:{
                    getDeviceUserId
                }
            }
        };
        const succCb = jest.fn();
        const errCb = jest.fn();
        createAsyncFun.default=jest.fn(
            (action,ERROR_MSG,resActionCreator,finCb)=>{
                finCb(fin,action,resActionCreator,succCb,errCb);
            }
        );
        await asyncs.getDeviceUserId(
            actions.getDeviceUserId({})
        );
        expect(getDeviceUserId).toMatchSnapshot();
        expect(succCb).toMatchSnapshot();
        expect(createAsyncFun).toMatchSnapshot();
    });

    it('getMonitorInfo',async ()=>{
        const getMonitorInfo = jest.fn((succCb,errCb)=>{
            succCb({});
        });
        const fin = {
            desktop:{
                System:{
                    getMonitorInfo
                }
            }
        };
        const succCb = jest.fn();
        const errCb = jest.fn();
        createAsyncFun.default=jest.fn(
            (action,ERROR_MSG,resActionCreator,finCb)=>{
                finCb(fin,action,resActionCreator,succCb,errCb);
            }
        );
        await asyncs.getMonitorInfo(
            actions.getMonitorInfo({})
        );
        expect(getMonitorInfo).toMatchSnapshot();
        expect(succCb).toMatchSnapshot();
        expect(createAsyncFun).toMatchSnapshot();
    });

    it('getVersion',async ()=>{
        const getVersion = jest.fn((succCb,errCb)=>{
            succCb();
        });
        const fin = {
            desktop:{
                System:{
                    getVersion
                }
            }
        };
        const succCb = jest.fn();
        const errCb = jest.fn();
        createAsyncFun.default=jest.fn(
            (action,ERROR_MSG,resActionCreator,finCb)=>{
                finCb(fin,action,resActionCreator,succCb,errCb);
            }
        );
        await asyncs.getVersion(
            actions.getVersion({})
        );
        expect(getVersion).toMatchSnapshot();
        expect(succCb).toMatchSnapshot();
        expect(createAsyncFun).toMatchSnapshot();
    });

    it('getHostSpecs',async ()=>{
        const getHostSpecs = jest.fn((succCb,errCb)=>{
            succCb({});
        });
        const fin = {
            desktop:{
                System:{
                    getHostSpecs
                }
            }
        };
        const succCb = jest.fn();
        const errCb = jest.fn();
        createAsyncFun.default=jest.fn(
            (action,ERROR_MSG,resActionCreator,finCb)=>{
                finCb(fin,action,resActionCreator,succCb,errCb);
            }
        );
        await asyncs.getHostSpecs(
            actions.getHostSpecs({})
        );
        expect(getHostSpecs).toMatchSnapshot();
        expect(succCb).toMatchSnapshot();
        expect(createAsyncFun).toMatchSnapshot();
    });

    it('getEnvironmentVariable',async ()=>{
        const getEnvironmentVariable = jest.fn((env,succCb,errCb)=>{
            succCb();
        });
        const fin = {
            desktop:{
                System:{
                    getEnvironmentVariable
                }
            }
        };
        const succCb = jest.fn();
        const errCb = jest.fn();
        createAsyncFun.default=jest.fn(
            (action,ERROR_MSG,resActionCreator,finCb)=>{
                finCb(fin,action,resActionCreator,succCb,errCb);
            }
        );
        await asyncs.getEnvironmentVariable(
            actions.getEnvironmentVariable({
                env:'env'
            })
        );
        expect(getEnvironmentVariable).toMatchSnapshot();
        expect(succCb).toMatchSnapshot();
        expect(createAsyncFun).toMatchSnapshot();
    });

    it('clearCache',async ()=>{
        const clearCache = jest.fn((options,succCb,errCb)=>{
            succCb();
        });
        const fin = {
            desktop:{
                System:{
                    clearCache
                }
            }
        };
        const succCb = jest.fn();
        const errCb = jest.fn();
        createAsyncFun.default=jest.fn(
            (action,ERROR_MSG,resActionCreator,finCb)=>{
                finCb(fin,action,resActionCreator,succCb,errCb);
            }
        );
        await asyncs.clearCache(
            actions.clearCache({})
        );
        expect(clearCache).toMatchSnapshot();
        expect(succCb).toMatchSnapshot();
        expect(createAsyncFun).toMatchSnapshot();
    });

})