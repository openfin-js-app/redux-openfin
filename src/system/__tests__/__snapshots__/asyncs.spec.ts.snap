// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`System asyncs clearCache 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {},
      [Function],
      [MockFunction],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs clearCache 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "error": undefined,
        "payload": Object {},
        "type": "REDUX_OPENFIN_RES::CLEAR_CACHE_RES",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs clearCache 3`] = `
Object {
  "__esModule": true,
  "default": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "error": undefined,
          "payload": Object {},
          "type": "REDUX_OPENFIN_REQ::CLEAR_CACHE",
        },
        "OpenFin API call System.clearCache() failed.",
        [Function],
        [Function],
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
}
`;

exports[`System asyncs getDeviceUserId 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Function],
      [MockFunction],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getDeviceUserId 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "error": undefined,
        "payload": Object {
          "id": undefined,
        },
        "type": "REDUX_OPENFIN_RES::GET_DEVICE_USER_ID_RES",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getDeviceUserId 3`] = `
Object {
  "__esModule": true,
  "default": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "error": undefined,
          "payload": Object {},
          "type": "REDUX_OPENFIN_REQ::GET_DEVICE_USER_ID",
        },
        "OpenFin API call System.getDeviceUserId() failed.",
        [Function],
        [Function],
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
}
`;

exports[`System asyncs getEnvironmentVariable 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "env",
      [Function],
      [MockFunction],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getEnvironmentVariable 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "error": undefined,
        "payload": Object {
          "env": "env",
          "value": undefined,
        },
        "type": "REDUX_OPENFIN_RES::GET_ENVIRONMENT_VARIABLE_RES",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getEnvironmentVariable 3`] = `
Object {
  "__esModule": true,
  "default": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "error": undefined,
          "payload": Object {
            "env": "env",
          },
          "type": "REDUX_OPENFIN_REQ::GET_ENVIRONMENT_VARIABLE",
        },
        "OpenFin API call System.getEnvironmentVariable() failed.",
        [Function],
        [Function],
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
}
`;

exports[`System asyncs getHostSpecs 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Function],
      [MockFunction],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getHostSpecs 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "error": undefined,
        "payload": Object {},
        "type": "REDUX_OPENFIN::GET_HOST_SPECS_RES",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getHostSpecs 3`] = `
Object {
  "__esModule": true,
  "default": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "error": false,
          "payload": Object {},
          "type": "REDUX_OPENFIN::GET_HOST_SPECS",
        },
        "OpenFin API call System.getHostSpecs() failed.",
        [Function],
        [Function],
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
}
`;

exports[`System asyncs getMachineId 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Function],
      [MockFunction],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getMachineId 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "error": undefined,
        "payload": Object {
          "id": undefined,
        },
        "type": "REDUX_OPENFIN_RES::GET_MACHINE_ID_RES",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getMachineId 3`] = `
Object {
  "__esModule": true,
  "default": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "error": undefined,
          "payload": Object {},
          "type": "REDUX_OPENFIN_REQ::GET_MACHINE_ID",
        },
        "OpenFin API call System.getMachine() failed.",
        [Function],
        [Function],
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
}
`;

exports[`System asyncs getMonitorInfo 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Function],
      [MockFunction],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getMonitorInfo 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "error": undefined,
        "payload": Object {},
        "type": "REDUX_OPENFIN::GET_MONITOR_INFO_RES",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getMonitorInfo 3`] = `
Object {
  "__esModule": true,
  "default": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "error": false,
          "payload": Object {},
          "type": "REDUX_OPENFIN::GET_MONITOR_INFO",
        },
        "OpenFin API call System.getMonitorInfo() failed.",
        [Function],
        [Function],
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
}
`;

exports[`System asyncs getVersion 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Function],
      [MockFunction],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getVersion 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "error": undefined,
        "payload": Object {
          "version": undefined,
        },
        "type": "REDUX_OPENFIN_RES::GET_VERSION_RES",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`System asyncs getVersion 3`] = `
Object {
  "__esModule": true,
  "default": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "error": undefined,
          "payload": Object {},
          "type": "REDUX_OPENFIN_REQ::GET_VERSION",
        },
        "OpenFin API call System.getVersion() failed.",
        [Function],
        [Function],
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
}
`;
